### PR TITLE
Fix/use webserver

### LINKED ISSUE #320: Expose webserver running status
With the latest branch https://github.com/performous/performous/tree/feature/PortInUseFix
We helped performous not to crash and ignore the webserver if the port is not free.
The text is a different matter as it still shows `Webserver active on 0.0.0.0:8080` 

We need some way to expose the running state of the webserver to be able to switch the message to for example: "Trying to start up webserver" and when it failed: "Failed to start webserver".

### LINKED ISSUE #334: Webserver code needs to be removed from screen_intro and disabled when not USE_WEBSERVER
Currently there is plenty of webserver-dependent code in screen_intro, for displaying current IP address and whatnot. This code runs even when webserver is entirely disabled (claiming that the server is active).

* Move all code other than drawing the text to webserver.cc
* Make it report the actual server status (i.e. did you actually manage to bind to a port) rather than config values

### PR DESCRIPTION
**Pull request by Baklap4**
### What does this PR do?
* Fixes the use of the `USE_WEBSERVER`
* Moved logic from screen_intro to webserver
* Using notification to pass messages through from one thread to another.

### Closes Issue(s)
* Closes #334 
* Closes #320 
### Motivation
The code of `screen_intro.cs` was cluttered. Therefore i moved the logic to webserver and introduced a messaging system so we can get messages accros to show the current webserver status.

### CODE REVIEW

File: game/screen.hh [hunk 1]
```diff
@@ -131,6 +135,9 @@ private:
 	AnimValue m_dialogTimeOut;
 	// Dialog members
 	std::unique_ptr<Dialog> m_dialog;
+#ifdef USE_WEBSERVER
+	std::string m_webserverMessage = "Trying to connect to webserver";
```
Comment (nieknooijens) [inline game/screen.hh]: good one!

**Comment by Baklap4** [comment]
why is semapthore constantly failing over ced? @Lord-Kamina

**Comment by Lord-Kamina** [comment]
I don't know, I  think it's a bug/error during checkout. Maybe you could change build commands so it checks if ced checked out properly and pull it again if it didn't?

Either way, usually if you manually trigger a rebuild it will work

**Comment by Baklap4** [comment]
@Lord-Kamina  updated this branch to contain latest master branch

**Comment by Baklap4** [comment]
Any particular reason for cloudflare dns instead of google's dns?
Ill change the fallbackport later this eve

**Comment by Lord-Kamina** [comment]
> Any particular reason for cloudflare dns instead of google's dns?

I for one have blocked google DNS on my router and it's a relatively common thing to do as some streaming services try forcing you to use them instead of whatever you have configured.

**Comment by nieknooijens** [comment]
> Any particular reason for cloudflare dns instead of google's dns?

cloudfare' s DNS is faster and has DNS over TLS support (so no eavesdropping)

**Comment by Baklap4** [comment]
Changed to cloudflare's dns and changed the logs message `fallbackport` to `fallback port`

**Comment by Lord-Kamina** [comment]
LGTM

**Pull request merged**